apps/studio/components/interfaces/Organization/BillingSettings/Restriction.tsx

```diff
@@ -0,0 +1,149 @@
+import Link from 'next/link'
+import { useOrgSettingsPageStateSnapshot } from 'state/organization-settings'
+import { AlertDescription_Shadcn_, AlertTitle_Shadcn_, Alert_Shadcn_, Button } from 'ui'
+
+export const Restriction = () => {
+  const org = useSelectedOrganization()
+  const { data: usage, isSuccess: isSuccessOrgUsage } = useOrgUsageQuery({ orgSlug: org?.slug })
+  const { data: subscription, isSuccess: isSuccessSubscription } = useOrgSubscriptionQuery({
+    orgSlug: org?.slug,
+  })
+  const snap = useOrgSettingsPageStateSnapshot()
+
+  const hasExceededAnyLimits = Boolean(
+    usage?.usages.find(
+      (metric) =>
+        !metric.unlimited && metric.capped && metric.usage > (metric?.pricing_free_units ?? 0)
+    )
+  )
+
+  // don't show any alerts until everything has been fetched
+  if (!isSuccessOrgUsage || !isSuccessSubscription || !org) {
+    return null
+  }
+
+  let shownAlert: 'exceededLimits' | 'gracePeriod' | 'gracePeriodOver' | 'restricted' | null = null
+
+  if (subscription && hasExceededAnyLimits && !org?.restriction_status) {
+    shownAlert = 'exceededLimits'
+  } else if (org?.restriction_status === 'grace_period') {
+    shownAlert = 'gracePeriod'
+  } else if (org?.restriction_status === 'grace_period_over') {
+    shownAlert = 'gracePeriodOver'
+  } else if (org?.restriction_status === 'restricted') {
+    shownAlert = 'restricted'
+  }
+
+  if (shownAlert === null) {
+    return null
+  }
+
+  return (
+    <>
+      {shownAlert === 'exceededLimits' && (
+        <Alert_Shadcn_ variant="destructive">
+          <AlertCircle strokeWidth={2} />
+          <div className="flex flex-row items-center justify-between">
+            <div>
+              <AlertTitle_Shadcn_>
+                Your organization's usage has exceeded its included quota
+              </AlertTitle_Shadcn_>
+              <AlertDescription_Shadcn_>
+                Your projects can become unresponsive or enter read-only mode.{' '}
+                {subscription.plan.id === 'free'
+                  ? 'Please upgrade to the Pro plan to ensure that your projects remain available.'
+                  : 'Please disable spend cap to ensure that your projects remain available.'}
+              </AlertDescription_Shadcn_>
+            </div>
+            <Button key="upgrade-button" asChild type="default">
+              <Link
+                href={`/org/${org?.slug}/billing?panel=${
+                  subscription.plan.id === 'free' ? 'subscriptionPlan' : 'costControl'
+                }`}
+              >
+                {subscription.plan.id === 'free' ? 'Upgrade plan' : 'Change spend cap'}
+              </Link>
+            </Button>
+          </div>
+        </Alert_Shadcn_>
+      )}
+      {shownAlert === 'gracePeriod' && (
+        <Alert_Shadcn_ variant="warning">
+          <AlertTriangle strokeWidth={2} />
+          <div className="flex flex-row items-center justify-between">
+            <div className="mr-3">
```

Comment:
apart from the copy changes I also added this sneaky "mr-3" to add more space between the text and the button. a poor man's way of doing this so feel free to change

Reply:
Replaced with `gap-3`, but otherwise ok.